src/libcore/str/lossy.rs

```diff
@@ -61,6 +60,7 @@ impl<'a> Iterator for Utf8LossyChunksIter<'a> {
         while i < self.source.len() {
             let i_ = i;
 
+            // SAFETY: 0 <= i < self.source.len()
```

Comment:
Well sure that's the contract required by `get_unchecked`, but why is that true here? It's the loop condition and the fact that `i` only gets increased, as far as I can tell.

Reply:
I elaborated a bit more here.